jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/HttpConnection.java

```diff
@@ -129,6 +129,12 @@ public HttpConnection(HttpConfiguration configuration, Connector connector, EndP
             LOG.debug("New HTTP Connection {}", this);
     }
 
+    @Override
+    public InvocationType getInvocationType()
+    {
+        return getServer().getInvocationType();
```

Comment:
Don't think this is right?
The same connection could be processed by different handlers, which have different invocation types.
Why do we even ask to a `HttpConnection` its invocation type?

Reply:
The InvocationType of the entire server is set when the server is started.   We can't have individual invocations for each individual request because:
1) we don't know who will process a request until we have called handle, by which time it is too late to go back to the execution strategy, unless we break up handle and process at the level of the managed selector.
2) handlers can be changed dynamically at run time, so if we do query the handlers for the invocation type, they will then race with any dynamic changes - unless we put in a global lock

Reply:
I have often wondered if ExecutionStrategy should work with `Supplier<Runnable>`s rather than just `Runnable`s.  That way, any task could produce another task that could be put back in the execution strategy, potentially with a different invocation type.
We'd then have the `HttpConnection` having a NON_BLOCKING type as a task that would call the `Handler.handle` methods to determine a Processor, which would then be returned from the first task as the next task and the invocationtype re-evaluated for the next dispatch.    But that is a big change, beyond the scope of the PR.